internal/app/machined/pkg/controllers/siderolink/manager.go

```diff
@@ -90,6 +93,12 @@ func (ctrl *ManagerController) Run(ctx context.Context, r controller.Runtime, lo
 				ID:        optional.Some(hardware.SystemInformationID),
 				Kind:      controller.InputWeak,
 			},
+			{
```

Comment:
this one is a bit tricky, as the controller restarts the SideroLink provisioning on each resource change, it might happen that it will be triggered for `MetaKey` update for `UniqueToken` while it previously already tried to connect without it.

I think the only meaningful way to handle that unfortunately is to have another intermediate resource which is managed by a controller which waits for MetaLoaded, and once Loaded either creates an empty intermediate resource with unique token, or fills it with MetaKey value.

Reply:
Can't we pass the state to the controller as we do with some of other controllers? That way we will not go for the `watch for input` behavior that we do not need here.

Another options are:
1. Add `Get` option that allows to bypass controller `Input`s.
2. Add special `Input` type that allows to read from namespace and type, but doesn't subscribe to it.

Reply:
neither of that helps here, I think. it's the logic: wait for META loaded, if machine token present, use it; if it changes, re-establish connection. It's pretty normal flow, there is no need for hacks, but we need to prepare proper Input for this controller in another controller, which will be better (separation of concerns)

Reply:
Added new resource and controller.